src/AbstractMultilineWhitespaceBeforeChainedCallsEndCharacterFixer.php

```diff
@@ -0,0 +1,274 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer;
+
+use PhpCsFixer\Fixer\ConfigurationDefinitionFixerInterface;
+use PhpCsFixer\Fixer\WhitespacesAwareFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Graham Campbell <graham@alt-three.com>
+ * @author Egidijus Girčys <e.gircys@gmail.com>
+ *
+ * @internal
+ */
+abstract class AbstractMultilineWhitespaceBeforeChainedCallsEndCharacterFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface, WhitespacesAwareFixerInterface
```

Comment:
if logic of 2 fixer is exactly the same but `$chainedCallsEndCharacter`, it proves that it's the very same fixer that shall have configuration - instead of having 2 artificial fixers

Reply:
Shall I deprecate `multiline_whitespace_before_semicolons` in favor of a new fixer (e.g. `multiline_whitespace_before_chained_call_end_characters`) with a `characters` option?

Reply:
is there a need for chained call indeed?
```php
$foo
    ->bar()
;
```

is not chained call ;)

Reply:
True, I'll rename things, but that doesn't answer my question :p

Reply:
I wanted to be sure first, as that was pretty damn big part of name ;)

yes, I would do the rename+proxy+deprecate trio

Reply:
PR reworked.